northstar/src/runtime/process/fs.rs

```diff
@@ -92,35 +93,34 @@ impl Mount {
 /// is referenced that does not exist.
 pub(super) async fn prepare_mounts(
     config: &Config,
-    container: &MountedContainer,
+    root: &Path,
+    manifest: Manifest,
 ) -> Result<(Vec<Mount>, Dev), Error> {
     let mut mounts = Vec::new();
     let mut dev = None;
-    let root = container
-        .root
-        .canonicalize()
-        .map_err(|e| Error::io("Canonicalize root", e))?;
 
-    mounts.push(proc(&root));
+    mounts.push(proc(root));
 
-    let manifest_mounts = &container.manifest.mounts;
+    let manifest_mounts = &manifest.mounts;
 
     for (target, mount) in manifest_mounts {
         match &mount {
             manifest::Mount::Bind(manifest::Bind { host, options }) => {
-                mounts.extend(bind(&root, target, host, options));
+                mounts.extend(bind(root, target, host, options));
             }
             manifest::Mount::Persist => {
-                mounts.push(persist(&root, target, config, container).await?);
+                let source = config.data_dir.join(manifest.name.to_string());
```

Comment:
We shouldn't forget to document that the version is intentionally not part of the path here. This allows upgrades with persistent data migration.

Reply:
Not sure I understand fully this but added your comment in a note above this line :blush: